widget/progressbarinfinite.go

```diff
@@ -94,42 +95,37 @@ func (p *infProgressRenderer) Objects() []fyne.CanvasObject {
 	return p.objects
 }
 
-func (p *infProgressRenderer) tickerSafe() *time.Ticker {
-	p.tickMutex.Lock()
-	defer p.tickMutex.Unlock()
-
-	return p.ticker
-}
-
 // Start the infinite progress bar background thread to update it continuously
 func (p *infProgressRenderer) start() {
-	if p.ticker == nil {
-		p.tickMutex.Lock()
+	if !p.running.Load().(bool) {
 		p.ticker = time.NewTicker(infiniteRefreshRate)
-		p.tickMutex.Unlock()
+		p.tickerStop = make(chan bool, 1)
+		p.running.Store(true)
 
 		go p.infiniteProgressLoop()
 	}
 }
 
 // Stop the infinite progress goroutine and sets value to the Max
 func (p *infProgressRenderer) stop() {
-	if p.tickerSafe() != nil {
-		p.tickMutex.Lock()
+	if p.running.Load().(bool) {
```

Comment:
This is a misuse of atomic.Value. It only guarantees that the `Load()` and the `Store()` calls are themselves atomic, not that the whole section between the `Load()` and `Store()` is atomic. You need to either compare-and-swap or, preferably, guard this section with a mutex.